### PR TITLE
Filesystem: Use Exists instead of ListDirectory to check whether a FilesystemView is valid

### LINKED ISSUE #2682: Vita crashes when opening a ZIP
(just to have this tracked)

### LINKED ISSUE #2683: Vita: Touch coordinates misaligned
via https://github.com/EasyRPG/Player/pull/2635#issuecomment-958297116

Just installed 0.7 on my Vita 1000, and it seems the touch area is misaligned on the vertical axis.
I mean, the top of the area is fine, but if I want to touch the bottom row buttons, I have to touch the middle of the scree

### PR DESCRIPTION
**Pull request by Ghabry**
ListDirectory receives in some cases an empty string here which will return the files of the working directory on most platforms and succeed.

On the Vita however this fails.

Fix #2682 
Fix #2683

### CODE REVIEW

**Comment by Ghabry** [comment]
Jenkins: Test this please

File: src/game_player.cpp [hunk 1]
```diff
@@ -215,7 +215,23 @@ void Game_Player::UpdateScroll(int amount, bool was_jumping) {
 		return;
 	}
 
-	Game_Map::Scroll(sx * amount, sy * amount);
+	int move_sx = 0;
```
Comment (carstene1ns) [inline src/game_player.cpp:218]: It looks like something was messed up here. github says this is a merge commit. Maybe rebase once more?

**Review by carstene1ns — APPROVED** [review]
Will do a final test on Wii, but LGTM.

**Comment by ttyborg** [comment]
Downloaded and tried the new build, both #2682 and #2683 seems to be fixed.

**Review by fdelapena — APPROVED** [review]

**Comment by Ghabry** [comment]
Jenkins: Test this please

**Comment by Ghabry** [comment]
Jenkins: Test this please

**Pull request merged**